### PR TITLE
fix(report): fix error with unmarshal of `ExperimentalModifiedFindings`

### LINKED ISSUE #7461: bug(convert): unable to decode ModifiedFinding.Results.ExperimentalModifiedFindings.Finding field
## Description
We use `finding` interface for `ModifiedFinding.Results.ExperimentalModifiedFindings.Finding field.
That is why we see error when converting json report with `ExperimentalModifiedFindings` field:
```
➜ trivy -d convert --show-suppressed  --exit-code 0 reports/trivyFull.json 
...
2024-09-09T10:22:22+06:00	FATAL	Fatal error
  - json decode error:
    github.com/aquasecurity/trivy/pkg/commands/convert.Run
        github.com/aquasecurity/trivy/pkg/commands/convert/run.go:32
  - json: cannot unmarshal object into Go struct field ModifiedFinding.Results.ExperimentalModifiedFindings.Finding of type types.finding

```

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Fix error with unmarshal of ExperimentalModifiedFindings (e.g. in `convert` mode)
  ```bash
  ➜ trivy -d convert --show-suppressed  --exit-code 0 reports/trivyFull.json 
  ...
  2024-09-09T10:22:22+06:00	FATAL	Fatal error
    - json decode error:
      github.com/aquasecurity/trivy/pkg/commands/convert.Run
          github.com/aquasecurity/trivy/pkg/commands/convert/run.go:32
    - json: cannot unmarshal object into Go struct field ModifiedFinding.Results.ExperimentalModifiedFindings.Finding of type types.finding

  ```  

## Related issues
- Close #7461

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
>We don't run MarshalJSON functions for ExperimentalModifiedFindings.Finding. e.g.:

This problem will be solved after https://github.com/aquasecurity/trivy-db/pull/436.

**Comment by knqyf263** [comment]
I'll also open another PR to fix this line.
https://github.com/aquasecurity/trivy/blob/5c37361600d922db27dd594b2a80c010a19b3a6e/pkg/fanal/types/package.go#L81

File: pkg/types/finding.go [hunk 1]
```diff
@@ -45,3 +51,96 @@ func NewModifiedFinding(f finding, status FindingStatus, statement, source strin
...
+	}{}
+
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+
+	m.Type = raw.Type
+	m.Status = raw.Status
+	m.Statement = raw.Statement
+	m.Source = raw.Source
```
Comment (knqyf263) [inline pkg/types/finding.go:113]: Can we refactor as below?

```
    type Alias ModifiedFinding
    aux := &struct {
        Finding json.RawMessage `json:"Finding"`
        *Alias
    }{
        Alias: (*Alias)(m),
    }
```

cf. https://github.com/aquasecurity/trivy/blob/5c37361600d922db27dd594b2a80c010a19b3a6e/pkg/fanal/types/package.go#L97-L121

Comment (DmitriyLewen) [reply pkg/types/finding.go]: Updated in https://github.com/aquasecurity/trivy/pull/7463/commits/06136214854575d0a54c411a8b1add802b24acf1

**Comment by DmitriyLewen** [comment]
@knqyf263 I merged #7483. So this PR is ready for review.

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@DmitriyLewen Do you think we want to backport this fix? I'm not sure this is a critical issue because this problem happens only when `--show-suppressed` is passed.

**Comment by DmitriyLewen** [comment]
hmm... I'm not sure about that
To reproduce this bug you need to use `json` with ExperimentalModifiedFindings into `convert` mode.

This is really quite rare case
We can just wait for feedback from users.
if many people encounter this - we will release a patch

**Comment by DmitriyLewen** [comment]
@aqua-bot backport release/v0.55

**Pull request merged**